### ISSUE TITLE
postgres_conf: Stacktrace on Postgres 10+ when using default path

### ISSUE THREAD

**Issue by eloyekunle** [inspec/inspec#3783]
### 🎛 Description

Bug Report

### 🌍 InSpec and Platform Version

Inspec 3.4.1 on Ubuntu 18.10.

### 🤔 Replication Case

With Postgres 11 installed, create a control:
```ruby
describe postgres_conf do
  its('port') {should eq '5432'}
end
```
And execute it. Stacktrace below.

If you specify a configuration path, you don't get an error and the test passes:
```ruby
describe postgres_conf('/etc/postgresql/11/main/postgresql.conf') do
  its('port') {should eq '5432'}
end
```

### 💻 Stacktrace

```
Traceback (most recent call last):
	27: from /usr/bin/inspec:122:in `<main>'
	26: from /usr/bin/inspec:122:in `load'
	25: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/bin/inspec:12:in `<top (required)>'
	24: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	23: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	22: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	21: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	20: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/cli.rb:258:in `exec'
	19: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/runner.rb:110:in `run'
	18: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/runner.rb:81:in `load'
	17: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/runner.rb:81:in `each'
	16: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/runner.rb:99:in `block in load'
	15: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile.rb:192:in `collect_tests'
	14: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile.rb:192:in `each'
	13: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile.rb:195:in `block in collect_tests'
	12: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile_context.rb:144:in `load_control_file'
	11: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile_context.rb:160:in `load_with_context'
	10: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/profile_context.rb:160:in `instance_eval'
	 9: from fucks.rb:1:in `load_with_context'
	 8: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/resource.rb:51:in `block (3 levels) in create_dsl'
	 7: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/resource.rb:51:in `new'
	 6: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/plugin/v1/plugin_types/resource.rb:96:in `initialize'
	 5: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/resources/postgres_conf.rb:26:in `initialize'
	 4: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/backend.rb:81:in `block (3 levels) in create'
	 3: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/backend.rb:81:in `new'
	 2: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/inspec/plugin/v1/plugin_types/resource.rb:96:in `initialize'
	 1: from /opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/resources/postgres.rb:20:in `initialize'
/opt/inspec/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.4.1/lib/resources/postgres.rb:118:in `cluster_from_dir': undefined method `chomp' for nil:NilClass (NoMethodError)
```

**Comment by eloyekunle** [comment]
I've not tested it but it seems the code was not designed to work with Postgres 10 & 11 because the inspec command picking out the current installed Postgres version [here](https://github.com/inspec/inspec/blob/master/lib/resources/postgres.rb#L68) is designed to work with Postgres versions < 10.
From version 10 onwards, Postgres only uses the major version to name its configuration directories.

From their [versioning documentation](https://www.postgresql.org/support/versioning/):

> Starting with PostgreSQL 10, a major version is indicated by increasing the first part of the version, e.g. 10 to 11. Before PostgreSQL 10, a major version was indicated by increasing either the first or second part of the version number, e.g. 9.5 to 9.6.

**Comment by clintoncwolfe** [comment]
Closed by #3833
